test/tests.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const request = require('superagent');
+const expect = require('chai').expect;
+require('../server.js');
+
+describe('Data routes', function() {
+  var data = null;
+
+  describe('POST: /api/ski-data', function() {
+    it('should return a data object', function(done) {
+      request.post('localhost:3000/api/ski-data')
+        .send({location: 'Mt baker', rating: 10})
+        .end((err, res) => {
+          if(err) return done(err);
+          expect(res.status).to.equal(200);
+          expect(res.body.location).to.equal('Mt baker');
+          expect(res.body.rating).to.equal(10);
+          data = res.body;
+          done();
+        });
+    });
+    it('should return a data object', function(done) {
+      request.post('localhost:3000/api/ski-data')
+        .send({location: 'mt baker'})
+        .end((res) => {
+          expect(res.status).to.equal(400);
+          done();
+        });
+    });
+  });
+  describe('GET: /api/ski-data', function() {
+    it('should return a data object', function(done) {
+      request.get(`localhost:3000/api/ski-data?id=${data.id}`)
+        .end((err, res) => {
+          if(err) return done(err);
+          expect(res.status).to.equal(200);
+          expect(res.body.location).to.equal('Mt baker');
+          expect(res.body.rating).to.equal(10);
+          done();
+        });
+    });
+    it('should return a 404 error', function(done) {
+      request.get('localhost:3000/api/ski-data?id=123456789')
+        .end((res) => {
+          expect(res.status).to.equal(404);
+          done();
+        });
+    });
+    it('should return a 400 error', function(done) {
+      request.get('localhost:3000/api/ski-data')
+        .end((res) => {
+          expect(res.status).to.equal(400);
+          done();
+        });
+    });
+  });
+  describe('DELETE: /api/ski-data', function() {
+    it('should delete a data object', function(done) {
+      request.delete(`localhost:3000/api/ski-data?id=${data.id}`)
+        .end((err, res) => {
+          if(err) return done(err);
+          expect(res.status).to.equal(204);
+          done();
+        });
+    });
+  });
+});
```

Comment:
The PORT you are having your server listen on should be the same PORT you are using for your tests. This way, if someone clones your code and runs gulp or mocha the tests will pass. For instance, these all failed on my machine until I changed them to the PORT your server is looking for. If you have PORT as an environmental variable in your .bashrc I would remove it.